### ISSUE TITLE
Add parallel backups

### ISSUE THREAD

**Issue by vy-let** [vy-let/yoink#5]
At the very least, backups from separate hosts can run in parallel, since they necessarily operate on disjoint sets of files. We need to have a config on the client specifying how many hosts to run at the same time.

As a separate issue, we may want to investigate the possibility of multiple parallel transfer sources per-host. Since we don't explicitly tie the side-effects of before-hooks to specific transfer areas, we'll want to make sure that no transfers start before pre-hooks are done, and that post-hooks don't start until all transfers are done.
